Google Ads: Update UI for Google Ads dashboard card

itsmeichigo opened the pull request:
<!-- Remember about a good descriptive title. -->

Part of #13230 
<!-- Id number of the GitHub issue this PR addresses. -->

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This PR updates the layout of the Google Ads dashboard card. Changes include:
- Updated the card view model:
  - Fetch last campaign's details. Since the REST API doesn't support pagination, we're fetching all campaigns and get the last active (enabled) one. If no active campaigns are found, we display the last disabled one.
  - Retrieve last campaign's stats in the background. I'm currently fetching all stats but asking the plugin team if we can fetch stats for a specific campaign only (p1720153987272339-slack-C02BB3F30TG).
- Added layout for campaign details and stats. I'm using the same layout with Blaze's for simplicity. Stats will not be displayed if they cannot be loaded.
- Handled error loading states.
- Handled tap on the dismiss button.
- Updated wording for the empty state.
- Added generic tracking events related to dynamic dashboard.

Due to the large size of the PR, unit tests and navigation for the card will be handled in a separate PR.

## Testing steps
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->
- Pre-requisite: Install, activate, and set up the Google Ads plugin to a test site following pe5sF9-2Qo-p2. Ensure that you're using the plugin version 2.7.5 or above.
- Build the app and log in to your store. 
- Enable Google Ads card if you haven't already. Confirm that the card shows the loading state when fetching data. Confirm that `dynamic_dashboard_card_data_loading_started` with `type: google-ads` is tracked.
- When data loading completed, confirm that `dynamic_dashboard_card_data_loading_completed` with `type: google-ads` is tracked
- If your store doesn't have any campaign, confirm that the empty state is displayed.
- Create a Google ads campaign for your store in wp-admin or from the Hub menu.
- Pull to refresh the dashboard screen and confirm that the campaign details are displayed.
- Disconnect the internet connection and reload the dashboard.
- Confirm that the error state is displayed and `dynamic_dashboard_card_data_loading_failed` with `type: google-ads` and error details is tracked.
- Tap on Retry, confirm that `dynamic_dashboard_card_retry_tapped` is tracked with `type: google-ads`.
- Tap on the ellipsis menu > Hide Google Ads. Confirm that Customize screen is displayed and `dynamic_dashboard_hide_card_tapped` is tracked with `type: google-ads`.

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->
Loading state | Empty state | Campaign details | Error state
--- | --- | --- | ---
<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/e157b0f1-1f9c-45ea-ad1d-8a2a8ed57538" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/76e4ac0b-69d3-4630-bc71-d557556bc4e2" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/1354c275-57dc-48d1-ae84-f54bc8a30570" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/3cb6de27-04e0-4b5c-9fe9-f8365a6e7491" width=320 />

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

itsmeichigo committed:
Handle dismissing google ads card

itsmeichigo committed:
Handle fetching last google ads campaign on dashboard card

itsmeichigo committed:
Prioritize showing last enabled campaign.

itsmeichigo committed:
Add extension for google ads campaign status for display

itsmeichigo committed:
Add detail view for Google ads campaign

itsmeichigo committed:
Display detail view on google dashboard card

itsmeichigo committed:
Update loading and error states

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

itsmeichigo committed:
Merge branch 'trunk' into feat/13230-campaign-details-on-gla-card

itsmeichigo committed:
Retrieve last campaign stats in the background

itsmeichigo committed:
Display stats on campaign details

itsmeichigo committed:
Filter stats results by campaign ID

itsmeichigo committed:
Update GoogleAdsCampaignCoordinator's property shouldStartCampaignCreation

itsmeichigo committed:
Navigation from dashboard card to google campaign

itsmeichigo committed:
Hide CTAs from error state

itsmeichigo committed:
Revert "Navigation from dashboard card to google campaign"

This reverts commit ca52d4f2c48aef646838a32a65b6f7d32fca6f0c.

itsmeichigo committed:
Revert "Update GoogleAdsCampaignCoordinator's property shouldStartCampaignCreation"

This reverts commit e5a5ee49da91e68cb67ac6d192d32542b85e8e40.

selanthiraiyan commented on WooCommerce/Classes/ViewRelated/Dashboard/GoogleAds/GoogleAdsCampaignDetailView.swift:
```diff
@@ -0,0 +1,135 @@
+                    )
+                    Text(campaign.name)
+                        .font(.headline)
+                        .fontWeight(.semibold)
```
Opinion: The headline style looks too big to me. I understand that this styling comes from the Blaze card.

<img width="416" alt="image" src="https://github.com/woocommerce/woocommerce-ios/assets/524475/f0363aa2-a92c-4e59-9092-d48835e51372">


If it helps, I took a look at the dynamic dashboard Figma file and noticed that it uses subheadline font for the campaign name. 

<img width="801" alt="image" src="https://github.com/woocommerce/woocommerce-ios/assets/524475/21d92cdc-eb3c-47e0-b823-f58f7d491162">

itsmeichigo replied:
Nice suggestion! Updated the font size in 82f5379ab6c48d807e6d5c94739b6b377b33ce1b.

selanthiraiyan replied:
Thanks, Huong! Should we also update the Blaze dashboard card view for consistency?

itsmeichigo replied:
Updated in 007db202ed970c2b2d2d24c03280d731edcd744a.

selanthiraiyan commented on WooCommerce/Classes/ViewRelated/Dashboard/GoogleAds/GoogleAdsCampaignDetailView.swift:
```diff
@@ -0,0 +1,135 @@
+    var body: some View {
+        VStack(spacing: Layout.contentSpacing) {
+            AdaptiveStack(horizontalAlignment: .leading, verticalAlignment: .center, spacing: Layout.contentSpacing) {
+                // campaign image
```
nit: Will we display a campaign image here in future instead of the google logo image?

itsmeichigo replied:
Google ads campaigns don't come with images so there's only google logo for now. We'll share the demo with the Google plugin team and discuss with them the possibility of introduce a new field for campaign image.

selanthiraiyan commented on WooCommerce/Classes/ViewRelated/Dashboard/GoogleAds/GoogleAdsCampaignDetailView.swift:
```diff
@@ -0,0 +1,135 @@
+                    Image(uiImage: .googleLogo)
+                        .resizable()
+                        .aspectRatio(contentMode: .fit)
+                        .frame(width: Layout.imageSize * scale, height: Layout.imageSize * scale)
```
Opinion: The logo size looks pretty large to me. It could be because the logo is rounded, and we don't use a corner radius modifier here.

itsmeichigo replied:
I think it's best that we keep the sizes and spacing consistent across different cards. I'll keep the image size as-is for now following Blaze card.

itsmeichigo committed:
Update description for empty state

selanthiraiyan commented on WooCommerce/Classes/ViewRelated/Dashboard/GoogleAds/GoogleAdsDashboardCardViewModel.swift:
```diff
@@ -29,6 +46,73 @@ final class GoogleAdsDashboardCardViewModel: ObservableObject {
+    }
+
+    @MainActor
+    func updateCampaignStats(campaignID: Int64) async {
```
Opinion: To treat this as a helper method, we could return the campaign stats total value from this method and assign it to `lastCampaignStats` from the calling place. I think this will improve readability.

itsmeichigo replied:
Good idea! Updated in 7acd489eaf9a4a40b000d589036947122e7a09c8.

selanthiraiyan reviewed (approved):
This works as expected. 🚀

itsmeichigo committed:
Update font size for campaign name

itsmeichigo committed:
Update logic for updateCampaignStats

itsmeichigo commented:
Thanks @selanthiraiyan for the reviews! I've addressed your comments and will enable auto-merge now. Please feel free to leave any other comments, I'll address them in a subsequent PR.

itsmeichigo committed:
Update font for Blaze campaign name
